Snowflake.Data/Core/SFDataConverter.cs

```diff
@@ -235,8 +236,10 @@ private static DateTimeOffset ConvertToDateTimeOffset(string srcVal, SFDataType
                     }
                     else
                     {
-                        DateTime srcDt = ((DateTime)srcVal);
-                        destVal = ((long)(srcDt.Subtract(UnixEpoch).Ticks * 100)).ToString();
+                        DateTime srcDt = (DateTime)srcVal;
+                        var diff = srcDt.Subtract(UnixEpoch);
+                        var tickDiff = diff.Ticks;
+                        destVal = $"{tickDiff}00"; // Cannot multiple tickDiff by 100 because long might overflow.

```

Comment:
Here overflow is avoided by not multiplying by 100 but adding trailing zeroes instead.